Improve error message for set! of imported identifier.

samth opened the pull request.

samth committed:
Improve error message for set! of imported identifier.

mflatt commented:
My hesitation here is whether "another module" is the right phrasing in a top-level context. I don't have a better idea, though, and maybe it's fine.

mfelleisen commented:
I'd use "different" rather than "another". Adding "or surrounding" might help the beginner. (Personally I think of module+ as different from the surrounding one.)

samth commented:
I think the best way to deal with that is to put the module where the identifier is defined in the error message, but I couldn't quite follow the error message protocol to add extra pieces.

sorawee commented:
Is this ready for merging?

samth commented:
It would really be better to do the thing I mentioned but this would be an improvement.

samth committed:
set!: name the defining module in the mutation error

Addresses review feedback on #4047: instead of the vague "another
module", report the specific module where the identifier is defined,
which is also accurate in a top-level context. Uses the same module
path index rendering as the other binding-to-module errors.
